gubernator.go

```diff
@@ -405,9 +405,23 @@ func (s *V1Instance) getGlobalRateLimit(ctx context.Context, req *RateLimitReq)
 		// Global rate limits are always stored as RateLimitResp regardless of algorithm
 		rl, ok := item.Value.(*RateLimitResp)
 		if ok {
+			// In the case we are not the owner, global behavior dictates that we respond with
+			// what ever the owner has broadcast to use as the response. However, in the case
+			// of TOKEN_BUCKET it makes little sense to wait for the owner to respond with OVER_LIMIT
+			// if we already know the remainder is 0. So we check for a remainder of 0 here and set
+			// OVER_LIMIT only if there are actual hits and this is not a RESET_REMAINING request and
+			// it's a TOKEN_BUCKET.
+			//
+			// We cannot preform this for LEAKY_BUCKET as we don't know how much time or what other requests
+			// might have influenced the leak rate at the owning peer.
+			// (Maybe we should preform the leak calculation here?????)
```

Comment:
As a newbie to this codebase, this comment suggests that we are breaking the single responsibility principle and we need to move this elsewhere. My thought was that we need a new API called `PeekRateLimit` that doesn't affect the hit count and only calculates the right status.

Reply:
100% agree with this, as I had the exact same thought. However, I'm planning on removing GRPC so I decided to wait for a future PR.

I could be convinced otherwise!

Reply:
I would avoid major refactors outside the context of this PR, for which I consider to be a bugfix. A follow up might make sense though.